airflow/jobs.py

```diff
@@ -224,15 +224,50 @@ def reset_state_for_orphaned_tasks(self, dag_run, session=None):
         # also consider running as the state might not have changed in the db yet
         running = self.executor.running
         tis = list()
-        tis.extend(dag_run.get_task_instances(state=State.SCHEDULED, session=session))
-        tis.extend(dag_run.get_task_instances(state=State.QUEUED, session=session))
+        tis.extend(dag_run.get_task_instances(state=[State.SCHEDULED, State.QUEUED],
+                   session=session))
 
         for ti in tis:
             if ti.key not in queued_tis and ti.key not in running:
                 self.logger.debug("Rescheduling orphaned task {}".format(ti))
                 ti.state = State.NONE
         session.commit()
 
+    @provide_session
+    def reset_state_for_orphaned_tasks_for_active_runs(self, limit=None, session=None):
+        """
+        This function resets the state for tasks in an active DagRun that are not known
+        by the executor so they will get scheduled again.
```

Comment:
Might be good to include a note why this doesn't simply use reset_state_for_orphaned_tasks_for_dag_run (for performance reasons?)